### ISSUE TITLE
Android build fail because of attribute "usesCleartextTraffic"

### ISSUE THREAD

**Issue by FreddyPoly** [Survicate/react-native-survicate#38]
Hello,

I'm currently trying to install the React Native SDK and I have trouble for Android. It builds fine for iOS.

---

### App settings

React Native : 0.64.4
Type of project : Metro
Tests have been made on a real device : HUAWEI HONOR 10 COL-L29
Android version on device : 8.1.0
Package @survicate/react-native-survicate : 1.1.18
Android configuration
```
buildToolsVersion = "29.0.3"
minSdkVersion = 21
compileSdkVersion = 33
targetSdkVersion = 31
ndkVersion = "20.1.5948944"
```
Node : v12.16.1
NPM : 6.13.4
```
compileOptions {
    sourceCompatibility JavaVersion.VERSION_1_8
    targetCompatibility JavaVersion.VERSION_1_8
}
```

---

I configured the settings

1. Added the WorkspaceKey to `AndroidManifest.xml`

2. Added the maven repository in `android/build.gradle`

When building for dev, I get the following error
```
Manifest merger failed : Attribute application@usesCleartextTraffic value=(true) from AndroidManifest.xml:7:18-53
        is also present at [com.survicate:survicate-sdk:1.7.4] AndroidManifest.xml:14:9-45 value=(false).
        Suggestion: add 'tools:replace="android:usesCleartextTraffic"' to <application> element at AndroidManifest.xml:7:5-9:19 to override.
```

If I add what the message suggests
```
<manifest xmlns:tools="http://schemas.android.com/tools"> // <-- I need this

    <application
      tools:replace="android:usesCleartextTraffic" // <-- here
      android:usesCleartextTraffic="true" // <-- and here
      ...
      >
      <meta-data android:name="com.survicate.surveys.workspaceKey" android:value="WORKSPACE_KEY"/>
      ...
    </application>
</manifest>
```

The building is successful but there is an error as soon as the app launches.
```
ReactNative: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.facebook.react.uimanager.FabricViewStateManager.hasStateWrapper()' on a null object reference
```

---

I created a repository to replicate the error, at : https://github.com/FreddyPoly/replicate-survicate-build-fail
This repo has only the demo app provided when creating a react native app for the first time. I initialized it with the typescript template

React Native : 0.70.6
Type of project : Metro
Tests have been made on a real device : HUAWEI HONOR 10 COL-L29
Android version on device : 8.1.0
Package @survicate/react-native-survicate : 1.1.18
Node : v16.14.0
NPM : 8.3.1
Android configuration :
```
buildToolsVersion = "31.0.0"
minSdkVersion = 21
compileSdkVersion = 31
targetSdkVersion = 31
```

### Steps to reproduce
- `npm i`
- `cd ios/ && pod install && cd ..`
- `cd android/ && ./gradlew clean && cd ..`
- `npx react-native start`
- `npx react-native run-android`

---

I also tried to change the version of the package "@survicate/react-native-survicate" to 1.1.17 and 1.1.15, but the same errors occured.

I need help with this error please.
Is there an option to install the survicate package without changing the manifest and adding 'tools:replace="android:usesCleartextTraffic"' ?

**Comment by j-brzozowski** [comment]
We have just released new version (1.1.19). We removed explicit setting of `usesCleartextTraffic` flag.

The error you mentioned seems to be a bug in React Native; please consider updating React Native version.

**Comment by FreddyPoly** [comment]
Hi

Thanks for your response. Indeed the error was caused by the React Native version. I updated to `0.70.6` and tried to install the new version `1.1.19`, and it builds fine on Android and iOS.

Thanks for your help. I close the issue
